### PR TITLE
Service implementation

### PR DESCRIPTION
**Pull request by alexkobal**
**Implemented:**
- Configuration window can be opened from the system tray
- Application can be closed from the system tray
- Keyboard shortcut binding
- Model structure for chunks
- Implemented auto typing functionality
- Copy to clipboard functionality
- Initialization with test shortcuts and text
- Showing extended error messages when a shortcut can't be assigned (throwing exception with extended error message)
- Refactoing of file structure
- Implemented speed setting for auto typing
- Added an option to cancel auto typing with a keyboard hook
- Implemented varying speed for auto typing

**TODO:**
- ~~Async auto typing (will not block the main thread of windows forms)~~ (Not needed with the current implementation)
- GUI
- [Settings](https://learn.microsoft.com/en-us/dotnet/desktop/winforms/advanced/application-settings-for-windows-forms?view=netframeworkdesktop-4.8)
- Communication between GUI and the model
- File I/O (XML)

### CODE REVIEW

File: AutoTyperGUI/Model/KeyboardHook/KeyboardHook.cs [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+        {
+            this.RegisterHotKey(modifier, key);
+        }
+
+        /// <summary>
+        /// Registers a hot key in the system.
+        /// </summary>
+        /// <param name="modifier">The modifiers that are associated with the hot key.</param>
+        /// <param name="key">The key itself that is associated with the hot key.</param>
+        public void RegisterHotKey(ModifierKeys modifier, Keys key)
```
Comment (alexkobal) [inline AutoTyperGUI/Model/KeyboardHook/KeyboardHook.cs:101]: Document that is throws exception

File: AutoTyperGUI/Model/Chunk.cs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+
+namespace AutoTyperGUI
+{
+    internal class Chunk
+    {
+        public string Text { get; set; }
+        public KeyboardHook ClipboardKHook { get; private set; }
```
Comment (alexkobal) [inline AutoTyperGUI/Model/Chunk.cs]: Handle unregistering/assigning new keyboard hooks

File: AutoTyperGUI/View/AutoTyperApplicationContext.cs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+            MenuItem exitMenuItem = new MenuItem("Exit", new EventHandler(Exit));
+
+            notifyIcon = new NotifyIcon();
+            notifyIcon.Icon = Properties.Resources.AppIcon;
+            notifyIcon.ContextMenu = new ContextMenu(new MenuItem[]
+                { configMenuItem, exitMenuItem });
+            notifyIcon.Visible = true;
+        }
+
+        ConfigurationWindow configWindow = new ConfigurationWindow();
```
Comment (alexkobal) [inline AutoTyperGUI/View/AutoTyperApplicationContext.cs:26]: Open Configuration window on the start of the application

File: AutoTyperGUI/Model/AutoTypeSettings.cs [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+using System.Text;
+using System.Threading.Tasks;
+
+namespace AutoTyperGUI
+{
+    internal class AutoTypeSettings
+    {
+        public TypingSpeed TypingSpeed { get; set; }
+        public double StdDeviation { get; set; }
+        public AutoTypeSettings() //TODO add constructor implementation instead of hardcoded values
```
Comment (alexkobal) [inline AutoTyperGUI/Model/AutoTypeSettings.cs]: Implement constructor

File: AutoTyperGUI/Model/KeyboardHook/KeyboardHook.cs (hunk 1 above)
Comment (alexkobal) [reply AutoTyperGUI/Model/KeyboardHook/KeyboardHook.cs:101]: DONE

File: AutoTyperGUI/Model/Chunk.cs (hunk 2 above)
Comment (alexkobal) [reply AutoTyperGUI/Model/Chunk.cs]: DONE

File: AutoTyperGUI/View/AutoTyperApplicationContext.cs (hunk 3 above)
Comment (alexkobal) [reply AutoTyperGUI/View/AutoTyperApplicationContext.cs:26]: DONE

File: AutoTyperGUI/Model/AutoTypeSettings.cs (hunk 4 above)
Comment (alexkobal) [reply AutoTyperGUI/Model/AutoTypeSettings.cs]: DONE

**Review by alexkobal — COMMENTED** [review]
All changes discussed on joint review are present in the last commit.
@gulya99 Please review the latest commit and approve if you have no additional findings.

**Review by gulya99 — APPROVED** [review]

**Pull request merged**